### ISSUE TITLE
Target vs Actuals IPTT: Export entire program with each target frequency in its own tab

### ISSUE THREAD

**Issue by jennyhavah** [mercycorps/TolaActivity#346]
# Story
When I want a comprehensive Target vs Actuals IPTT report in Excel format, it is an easy single-click process to export all of my indicator data, with data organized by target frequency.

# Acceptance criteria

## Accessing the "Export entire program" option
![image](https://user-images.githubusercontent.com/33670923/55089514-c1d18500-506a-11e9-9a38-708c288d297d.png)
- [x] 1. When I run a Target vs Actuals IPTT report on any target frequency, and then click the "Excel" button, I get a pop-up similar to the "Pin" one, with two options: "Current view" and "All program data"
- [x] 2. When I click "Current view", I download the same Excel export available on Production, with only the target frequency I am currently viewing.
     - [x] 2.1. Regression check: Start and End target period adjustments are maintained in the export.
     - [x] 2.2. Regression check: "Most recent" selections are maintained in the export.
     - [x] 2.3. Regression check: The default sort in the "All program data" export mirrors the default sort in the web view: Level, then number, then name.
- [x] 3. When I click "All program data", I download all target frequencies in an Excel workbook
     - [x] 3.1. Any target period adjustments in the current view are ignored. I get all target periods for every frequency.
     - [x] 3.2. Any "Most recent" selections in the current view are ignored. By definition "all program data" means "Show all" for every target frequency.
     - [x] 3.3. The default sort in the "All program data" export mirrors the default sort in the web view: Level, then number, then name.

## Workbook organization
- [x] 1. There is a tab for each target frequency, only for those target frequencies used in the program. (For example, if there are no semi-annual indicators in the program, there is no Semi-annual tab in the "All program data" workbook.) go in the following order, and are named a
- [x] 2. Tab order mimics the order of options the Target periods menu, exactly.
- [x] 3. Tab labels mimic the wording in the Target periods menu, exactly:
    1. Life of Program (LoP) only
     2. Midline and endline
     3. Annual
     4. Semi-annual
     5. Tri-annual
     6. Quarterly
     7. Monthly

## Open question up for discussion
Can we / do we want to preserve filtering of levels, types, sectors, sites, and indicators in the "All program data" workbook export?

**Comment by jennyhavah** [comment]
If you're wondering about that Changelog tab, see https://learn.mercycorps.org/index.php/TOLA:Partner_Access#Design_Phase_2_.28for_internal_development.29:_Fancy_change_log

**Comment by jennyhavah** [comment]
@camgnostic I just remembered something: With Partner Access, we are implementing a change log that is exportable from the IPTT via a button that looks something like this -- you may be able to see it on Dev2 now:

![image](https://user-images.githubusercontent.com/33670923/55113221-c19dad80-509b-11e9-9c52-cfd7075f3e27.png)

This change log is for the entire program and is usually included in an IPTT workbook. I'd like to include it in the export as the final tab. 

Question: Should I create a separate ticket for this, so that this one isn't blocked by it?

**Comment by camgnostic** [comment]
> Can we / do we want to preserve filtering of levels, types, sectors, sites, and indicators in the "All program data" workbook export?

We can, definitely.  It seems counter-intuitive to me to drop the filtering by periods but then preserve the other filters.  It also seems to belie the "All" in the "All program data" label.  Let me know if you want this feature added!

**Comment by camgnostic** [comment]
>Question: Should I create a separate ticket for this, so that this one isn't blocked by it?
Yes, this should be a separate ticket as it won't be available for me to work on in this branch until PA is fully merged in

**Comment by jennyhavah** [comment]
> We can, definitely. It seems counter-intuitive to me to drop the filtering by periods but then preserve the other filters. It also seems to belie the "All" in the "All program data" label. Let me know if you want this feature added!

That is a great point @camgnostic. I was thinking of "All" as in "All available time... for whatever indicators you want" but I think it makes more sense that "All means All". Never mind on that one!

**Comment by jennyhavah** [comment]
@camgnostic Looks like the gray RF level rows are not exporting. If I see those rows in the IPTT in the web view, I expect to see them in the export.

**Comment by jennyhavah** [comment]
The basic export functionality is working as intended. There are some issues related to filtering and exports that are covered in other tickets.
